File: backend/domains/auth/handlers/login.handler.go
package handlers

import (
	"context"

	"github.com/cloudwego/hertz/pkg/app"
	"github.com/erweixin/go-genai-stack/backend/domains/auth/http/dto"
)

// LoginHandler 用户登录 Handler
//
// 对应用例：Login
// HTTP Method: POST
// HTTP Path: /api/auth/login
// 认证：不需要
//
// Handler 职责（瘦层）：
//   - 解析 HTTP 请求
//   - 转换 DTO（使用转换层）
//   - 调用 Domain Service
//   - 转换响应（使用转换层）
func (deps *HandlerDependencies) LoginHandler(ctx context.Context, c *app.RequestContext) {
	// 1. 解析请求
	var req dto.LoginRequest
	if err := c.BindAndValidate(&req); err != nil {
		handleValidationError(c, err)
		return
	}

	// 2. 转换为 Domain Input（使用转换层）
	input := toLoginInput(req)

	// 3. 调用 Domain Service
	output, err := deps.authService.Login(ctx, input)
	if err != nil {
		handleDomainError(c, err)
		return
	}

	// 4. 转换为 HTTP 响应（使用转换层）
	c.JSON(200, toLoginResponse(output))
}

File: backend-nodejs/domains/auth/handlers/login.handler.ts
/**
 * Login Handler
 */

import type { FastifyRequest, FastifyReply } from 'fastify';
import type { HandlerDependencies } from './dependencies.js';
import type { LoginRequest } from '../http/dto/auth.js';
import { toLoginInput, toLoginResponse } from './converters.js';
import { createContextFromRequest } from '../../../shared/types/context.js';

/**
 * Login Handler
 * HTTP 适配层：处理用户登录的 HTTP 请求
 * 错误由全局错误处理中间件统一处理
 */
export async function loginHandler(
  deps: HandlerDependencies,
  req: FastifyRequest<{ Body: LoginRequest }>,
  reply: FastifyReply
): Promise<void> {
  const input = toLoginInput(req.body);
  const ctx = createContextFromRequest({ ...req });
  const output = await deps.authService.login(ctx, input);

  reply.code(200).send(toLoginResponse(output));
}
